contracts/token/GovernanceToken.sol

```diff
@@ -14,24 +14,23 @@ contract GovernanceToken is IERC165, SaleRounds {
     bytes4 public constant IID_IERC165 = type(IERC165).interfaceId;
 
     uint8 private decimalUnits;
-    uint constant private MAX_UINT256 = type(uint256).max;
 
     constructor(uint256 _initialAmount, string memory _tokenName,
                 uint8 _decimalUnits, string memory _tokenSymbol,
-                address _gameOwnerAddress)
-                SaleRounds(_tokenName, _tokenSymbol, _initialAmount, _decimalUnits, _gameOwnerAddress) {
+                address _gameOwnerAddress, address[] memory _walletAddresses)
+                SaleRounds(_tokenName, _tokenSymbol, _initialAmount, _decimalUnits, _gameOwnerAddress, _walletAddresses) {
         decimalUnits = _decimalUnits;
     }
 
-    function decimals() public view virtual override returns (uint8) {
-        return decimalUnits;
-    }
-
     function isERC20() external view returns (bool) {
         return address(this).supportsInterface(IID_IERC20);
     }
 
     function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
         return interfaceId == IID_IERC20 || interfaceId == IID_IERC165;
     }
+
+    function decimals() public view virtual override returns (uint8) {
+        return decimalUnits;
```

Comment:
We should think whether 18 decimals is useful at all, maybe 2 or 3 decimals would be better.

Reply:
Let's leave it at 18 for now.